blocks/monitoring/evaluators.py

```diff
@@ -238,4 +290,12 @@ def evaluate(self, data_stream):
                 'Only data independent variables were given,'
                 'will not iterate the over data!')
 
+        if self._accumulate_fun2 is not None:
+            for batch in data_stream.get_epoch_iterator(as_dict=True):
```

Comment:
You can loop only once over the data stream. Updating the non-Theano monitored quantities should be done in `process_batch`.